department-of-veterans-affairs/va.gov-cms
# VACMS-17115: Adds ECA and friends to the platform.

## Description

Relates to #17115

## Testing done
ECA Models can be created.

## Screenshots
![Screenshot 2024-02-16 at 11 57 33 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/2b7f7d3c-d559-4fa8-b683-075d5ff5210c)

## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

Verify ECA Works as Expected

As an administrator
1. Visit /admin/config/workflow/eca
   - [x] Validate that there are no existing Models
2. Then click the Add New Model button
3. Provide a name and click Save
4. On the next page, under 'Add event'
   - [x] Validate that there is an Event 'ECA cron event (ECA Base)' item
5. Select `ECA cron event (ECA Base)` and click 'Add Event'
6. On the subsequent page, click Save after making no changes.
7. Under the Action accordion on the next page
   - [x] Validate there is a 'Views: Execute query (ECA Views)' action
 
### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
